### PR TITLE
fix: adding `PeerDependenciesMeta` to `NpmLockfile` to fix issue with `turbo prune`

### PR DESCRIPTION
**Pull request by derekaug**
When unmarshalling the package-lock.json, since `PeerDependenciesMeta` is not present on the `NpmPackage` struct it gets dropped when `turbo prune` does its thing. This caused `npm ci` to complain due to the package and lock-file being out of sync as described in #2739 

By adding this property to the struct, it gets preserved when unmarshalling and persists in the lockfile after the prune, resulting in a pruned package and lock-file that remain in sync according to `npm`.

### CODE REVIEW

**Review by mehulkar — COMMENTED** [review]
This makes sense to me, but I want to defer to @chris-olszewski. 

Is it possible to add just a unit test to show that the marshalling works, or better yet, produce the failure you're trying to fix?

**Comment by derekaug** [comment]
> This makes sense to me, but I want to defer to @chris-olszewski.
> 
> Is it possible to add just a unit test to show that the marshalling works, or better yet, produce the failure you're trying to fix?

Should be possible, just need to find the best way to add the case.

File: cli/internal/lockfile/npm_lockfile.go [hunk 1]
```diff
@@ -44,6 +44,9 @@ type NpmPackage struct {
 	Dependencies         map[string]string `json:"dependencies,omitempty"`
 	DevDependencies      map[string]string `json:"devDependencies,omitempty"`
 	PeerDependencies     map[string]string `json:"peerDependencies,omitempty"`
+	PeerDependenciesMeta map[string]struct {
+		Optional bool `json:"optional"`
```
Comment (chris-olszewski) [inline cli/internal/lockfile/npm_lockfile.go:48]: I think you want to only want entries if `optional` is set to `true`. (Probably unimportant as npm is fairly relaxed with the exact representation in the lockfile)
```suggestion
		Optional bool `json:"optional,omitempty"`
```

**Review by chris-olszewski — COMMENTED** [review]
First of all, thanks for fixing this! This looks correct to me. I'll stamp once there's a unit test just verifying that data can make the round trip. Something as simple as unmarshalling and marshalling back one of the entries with `peerDependenciesMeta` set would be enough e.g.

```json
{
      "version": "13.0.0",
      "resolved": "https://registry.npmjs.org/eslint-config-next/-/eslint-config-next-13.0.0.tgz",
      "integrity": "sha512-y2nqWS2tycWySdVhb+rhp6CuDmDazGySqkzzQZf3UTyfHyC7og1m5m/AtMFwCo5mtvDqvw1BENin52kV9733lg==",
      "dependencies": {
        "eslint-plugin-react": "^7.31.7",
        "eslint-plugin-react-hooks": "^4.5.0"
      },
      "peerDependencies": {
        "eslint": "^7.23.0 || ^8.0.0",
        "typescript": ">=3.3.1"
      },
      "peerDependenciesMeta": {
        "typescript": {
          "optional": true
        }
      }
    }
```

File: cli/internal/lockfile/npm_lockfile.go (hunk 1 above)
Comment (derekaug) [reply cli/internal/lockfile/npm_lockfile.go]: I wondered about that, thanks for the suggestion!

**Comment by derekaug** [comment]
> First of all, thanks for fixing this! This looks correct to me. I'll stamp once there's a unit test just verifying that data can make the round trip. Something as simple as unmarshalling and marshalling back one of the entries with `peerDependenciesMeta` set would be enough e.g.
> 
> ```json
> {
>       "version": "13.0.0",
>       "resolved": "https://registry.npmjs.org/eslint-config-next/-/eslint-config-next-13.0.0.tgz",
>       "integrity": "sha512-y2nqWS2tycWySdVhb+rhp6CuDmDazGySqkzzQZf3UTyfHyC7og1m5m/AtMFwCo5mtvDqvw1BENin52kV9733lg==",
>       "dependencies": {
>         "eslint-plugin-react": "^7.31.7",
>         "eslint-plugin-react-hooks": "^4.5.0"
>       },
>       "peerDependencies": {
>         "eslint": "^7.23.0 || ^8.0.0",
>         "typescript": ">=3.3.1"
>       },
>       "peerDependenciesMeta": {
>         "typescript": {
>           "optional": true
>         }
>       }
>     }
> ```

I added a test case that should show that unmarshalling / marshalling is working with npm lockfiles. Not sure it's exactly what you were expecting, or proper. Beginner Go dev here so please have some mercy. 😅

**Review by chris-olszewski — APPROVED** [review]
The test looks great to me! Thanks for fixing this. I'll get it merged in once CI passes

**Pull request merged**